hop/subscribe.py

```diff
@@ -72,12 +113,20 @@ def _main(args=None):
     # set offset
     start_offset = "earliest" if args.earliest else "latest"
 
+    # set timeout
+    t = args.timeout
+    timeout = None if t == "-1" else float(t)
```

Comment:
Why was the `type=float` option removed from the timeout arg? I don't think you'll need to do the float casting or handling of -1 as a string with that in place.

Reply:
Good catch -- I removed `type` so that the user could specify `None` for indefinite timeout, but changed this to `-1`. Adding `type` back in.